src/network/network.cpp

```diff
@@ -1303,6 +1352,57 @@ void Network::FreeStringIds()
 	}
 }
 
+void Network::Client_Send_RESENDMAP()
+{
+	char str_desync[256];
+	format_string(str_desync, STR_MULTIPLAYER_DESYNC, NULL);
+	window_network_status_open(str_desync);
+
+	// send view information for the savefile
+	rct_window* w = window_get_main();
+	rct_viewport* viewport = w->viewport;
+
+	sint16 viewX, viewY, viewZoom, viewRotation;
+
+	viewX = viewport->view_width / 2 + viewport->view_x;
+	viewY = viewport->view_height / 2 + viewport->view_y;
+
+	viewZoom = viewport->zoom;
+	viewRotation = get_current_rotation();
+
+	std::unique_ptr<NetworkPacket> packet = std::move(NetworkPacket::Allocate());
+	*packet << (uint32)NETWORK_COMMAND_RESENDMAP << viewX << viewY << viewZoom << viewRotation;
+
+	server_connection.QueuePacket(std::move(packet));
+}
+
+void Network::Server_Handle_RESENDMAP(NetworkConnection& connection, NetworkPacket& packet)
+{
+	if (connection.GetLastResyncTime() - SDL_GetTicks() < NETWORK_RESYNC_TIMEOUT) {
```

Comment:
Which part specifically? Most of this code is for ensuring viewport persistence.